libethereum/Client.cpp

```diff
@@ -764,7 +764,7 @@ Block Client::block(h256 const& _blockHash, PopulationStatistics* o_stats) const
 	{
 		ex << errinfo_block(bc().block(_blockHash));
 		onBadBlock(ex);
-		return Block(chainParams().accountStartNonce);
+		return Block(bc());
```

Comment:
the `asOf` function (line 120) which has been replaced by `block(...)` also return `Block(bc());`.
the function you're talking about take one more parameter ( `PopulationStatistics* o_stats )` but is basically the same. so they should return the same thing in case of failure.